src/main/java/com/nvidia/spark/rapids/jni/GpuTimeZoneDB.java

```diff
@@ -0,0 +1,313 @@
+/*
+* Copyright (c) 2023, NVIDIA CORPORATION.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package com.nvidia.spark.rapids.jni;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.zone.ZoneOffsetTransition;
+import java.time.zone.ZoneRules;
+import java.time.zone.ZoneRulesException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+import java.util.concurrent.*;
+
+import ai.rapids.cudf.ColumnVector;
+import ai.rapids.cudf.DType;
+import ai.rapids.cudf.HostColumnVector;
+import ai.rapids.cudf.Table;
+
+public class GpuTimeZoneDB {
+
+  public static final int TIMEOUT_SECS = 300;
+
+
+  // For the timezone database, we store the transitions in a ColumnVector that is a list of 
+  // structs. The type of this column vector is:
+  //   LIST<STRUCT<utcInstant: int64, localInstant: int64, offset: int32>>
+  private CompletableFuture<Map<String, Integer>> zoneIdToTableFuture;
+  private CompletableFuture<HostColumnVector> fixedTransitionsFuture;
+
+  private boolean closed = false;
+
+  GpuTimeZoneDB() {
+    zoneIdToTableFuture = new CompletableFuture<>();
+    fixedTransitionsFuture = new CompletableFuture<>();
+  }
+  
+  private static GpuTimeZoneDB instance = new GpuTimeZoneDB();
+  static GpuTimeZoneDB getInstance() {
+    return instance;
+  }
+  
+  /**
+   * Start to cache the database. This should be called on startup of an executor. It should start
+   * to cache the data on the CPU in a background thread. It should return immediately and allow the
+   * other APIs to be called. Depending on what we want to * do we can have the other APIs block
+   * until this is done caching, or we can have private APIs that would let us load and use specific
+   * parts of the database. I prefer the former solution at least until we see a performance hit
+   * where we are waiting on the database to finish loading.
+   */
+  public static void cacheDatabase() {
+    if (!instance.isLoaded()) {
+      Executor executor = Executors.newSingleThreadExecutor(
```

Comment:
Will this threadsafe? Should we add to the function the `synchronized` qualifier?